### PR TITLE
[14.0] [MIG] mail_tracking_mailgun: Migration to v14.0

### PR DESCRIPTION
**Pull request by CarlosRoca13**
cc @Tecnativa TT30286

please @ernestotejeda @carlosdauden review this

### CODE REVIEW

File: mail_tracking_mailgun/models/ir_mail_server.py [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+from odoo import models
+
+
+class IrMailServer(models.Model):
+    _inherit = "ir.mail_server"
+
+    def _tracking_headers_add(self, tracking_email_id, headers):
+        headers = super(IrMailServer, self)._tracking_headers_add(
+            tracking_email_id, headers
+        )
```
Comment (pedrobaeza) [inline mail_tracking_mailgun/models/ir_mail_server.py]:
```suggestion
        headers = super()._tracking_headers_add(tracking_email_id, headers)
```

File: mail_tracking_mailgun/models/mail_tracking_event.py [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+
+class MailTrackingEvent(models.Model):
+    _inherit = "mail.tracking.event"
+
+    mailgun_id = fields.Char(string="Mailgun Event ID", copy="False", readonly=True)
+
+    def _process_data(self, tracking_email, metadata, event_type, state):
+        res = super(MailTrackingEvent, self)._process_data(
+            tracking_email, metadata, event_type, state
+        )
```
Comment (pedrobaeza) [inline mail_tracking_mailgun/models/mail_tracking_event.py]:
```suggestion
        res = super()._process_data(tracking_email, metadata, event_type, state)
```

**Review by pedrobaeza — APPROVED** [review]

**Comment by CarlosRoca13** [comment]
@pedrobaeza changes done

File: mail_tracking_mailgun/models/mail_tracking_email.py [hunk 3]
```diff
@@ -0,0 +1,273 @@
...
+
+    def _event_is_from_mailgun(self, event):
+        event = event or {}
+        return all([k in event for k in self._mailgun_mandatory_fields])
+
+    @api.model
+    def event_process(self, request, post, metadata, event_type=None):
+        res = super(MailTrackingEmail, self).event_process(
+            request, post, metadata, event_type=event_type
+        )
```
Comment (victoralmau) [inline mail_tracking_mailgun/models/mail_tracking_email.py]:
```suggestion
        res = super().event_process(
            request, post, metadata, event_type=event_type
        )
```

File: mail_tracking_mailgun/models/res_partner.py [hunk 4]
```diff
@@ -0,0 +1,196 @@
...
+from odoo.exceptions import UserError
+
+
+class ResPartner(models.Model):
+    _inherit = "res.partner"
+
+    def email_bounced_set(self, tracking_emails, reason, event=None):
+        res = super(ResPartner, self).email_bounced_set(
+            tracking_emails, reason, event=event
+        )
```
Comment (victoralmau) [inline mail_tracking_mailgun/models/res_partner.py]:
```suggestion
        res = super().email_bounced_set(
            tracking_emails, reason, event=event
        )
```

File: mail_tracking_mailgun/tests/test_mailgun.py [hunk 5]
```diff
@@ -0,0 +1,433 @@
...
+        )
+        mail.send()
+        # Search tracking created
+        tracking_email = self.env["mail.tracking.email"].search(
+            [("mail_id", "=", mail.id)]
+        )
+        return mail, tracking_email
+
+    def setUp(self):
+        super(TestMailgun, self).setUp()
```
Comment (victoralmau) [inline mail_tracking_mailgun/tests/test_mailgun.py]:
```suggestion
        super().setUp()
```

**Review by victoralmau — APPROVED** [review]
Please purge translations.

**Comment by CarlosRoca13** [comment]
@victoralmau changes done

**Pull request merged**